php/class-fieldmanager-media.php

```diff
@@ -154,22 +154,69 @@ public function presave( $value, $current_value = array() ) {
 	public function form_element( $value = array() ) {
 		if ( is_numeric( $value ) && $value > 0 ) {
 			$attachment = get_post( $value );
-			if ( strpos( $attachment->post_mime_type, 'image/' ) === 0 ) {
-				$preview  = esc_html( $this->selected_image_label ) . '<br />';
-				$preview .= '<a href="#">' . wp_get_attachment_image(
-					$value,
-					$this->preview_size,
-					false,
-					array(
-						'class' => $this->thumbnail_class,
-					)
-				) . '</a>';
-			} else {
-				$preview  = esc_html( $this->selected_file_label ) . '&nbsp;';
-				$preview .= wp_get_attachment_link( $value, $this->preview_size, true, true, $attachment->post_title );
+			// Open the preview wrapper.
+			$preview = '<div class="media-file-preview">';
+			$file_label = ''; // The uploaded file label - image or file.
+			$dashicon_class = '';
+
+			if ( $attachment->post_mime_type ) {
+				$preview .= '<a href="#">';
+				$image_type = strpos( $attachment->post_mime_type, 'image/' );
+
+				if ( strpos( $attachment->post_mime_type, 'audio/' ) === 0 ) {
+					$dashicon_class = 'dashicons-media-audio';
+				} elseif ( strpos( $attachment->post_mime_type, 'video/' ) === 0 ) {
+					$dashicon_class = 'dashicons-media-video';
+				} else {
+					$dashicon_class = 'dashicons-media-document';
+				}
+
+				$dashicon_class = apply_filters( 'fieldmanager_media_preview_icon', $dashicon_class, $attachment->post_mime_type );
```

Comment:
This new filter should get filter docs.

Also noting in passing that I'm not sure there would be a way to target the icon for a specific field or a specific attachment with this filter. I'm also not sure what use cases there would be for it. All the same, it might be worth keeping in the back of your head how such abilities might be provided by either new field properties or additional context passed to the filter.

Reply:
@dlh01 This filter was created so long ago that I can't recall the reasoning behind creating it. That said I am not sure this filter would provide that much value. Even if one was able to target for a specific field or attachment, would someone go through the effort in providing a different dashicon icon for an attachment? I am unsure and think that the filter should be removed unless there is a demand for such a feature.

Reply:
I like the approach of removing the filter 👍